instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/middlewares/tracer_middleware.rb

```diff
@@ -27,6 +27,7 @@ def trace_response(env)
 
             span.set_attribute('http.route', env['sinatra.route'].split.last) if env['sinatra.route']
             span.name = env['sinatra.route'] if env['sinatra.route']
+            span.record_exception(env['sinatra.error']) if env['sinatra.error']
```

Comment:
Based on the screen shots from your example, it looks like Sinatra reports HTTP 404 `Sinatra::NotFound` in `sinatra.error`. 

I _think_ this code should only record an exception for 5xx responses or exceptions that would result in a 500 error. 

Similar to what `dd-trace` does: https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/tracing/contrib/sinatra/tracer_middleware.rb#L78

Reply:
I really appreciate the time you are taking to look into addressing my concerns. 

> Long story short, this PR now only adds an Exception event, but does not mark the span as in Error; see below screenshot.

I do not think we should record exceptions for non-exceptional conditions that result in 4xx range _client_ errors (or any other ranges 1xx-3xx). 

I am a bit firm on _only_ using `span.record_exception` in cases where there is a _server side error_ i.e. 5xx range because they are issues related to the Sinatra application service that we are tracing. _Any objections or affirmations from the @open-telemetry/ruby-contrib-approvers?_

I do not have a lot of familiarity with Sinatra so I am afraid I do not have any recommendations other than digging into how ddtrace is able to disambiguate an exception that resulted in a server side error.

Reply:
thanks @arielvalentin and @ahayworth for your feedback.
Using a `404` for illustrating that we can add exceptions without marking the span as erroneous probably wasn't the best idea.

Anyway; https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/133/commits/edacd694e237d433446fba5f63dc88cc9205b664 fixes this, such that an exception and marking the span as erroneous is only done when there is a server error >500.